core/src/hash_table/pair.rs

```diff
@@ -10,40 +11,44 @@ pub struct Pair {
 
 impl Pair {
     /// build a new Pair from a chain and entry
+    ///
     /// Header is generated automatically
+    ///
     /// a Pair is immutable, but the chain is mutable if chain.push() is used.
+    ///
     /// this means that if two Pairs X and Y are generated for chain C then Pair X is pushed onto
     /// C to create chain C' (containing X), then Pair Y is no longer valid as the headers would
     /// need to include X. Pair Y can be regenerated with the same parameters as Y' and will be
     /// now be valid, the new Y' will include correct headers pointing to X.
+    ///
+    /// panics if entry is somehow invalid
```

Comment:
I would put this under the `# Panics` section

Reply:
There are several other cases of adding a panics documentation comment. Do you think they should all be in `# Panics` sections?

Reply:
Yes, definitely. Most of the `std` documentation follows this pattern